### PR TITLE
Added data pull progress bar

### PR DESCRIPTION
**Pull request by PatrickWaweru**
- Added a deterministic progress bar to indicate data pull progress with percentage indicator and records done over total records
- Plus some code optimizations
- Removed some space above the buttons

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/kenyaemrml/util/MLDataExchange.java [hunk 1]
```diff
@@ -401,4 +402,22 @@ public boolean getContinuePullingData() {
...
+	 * 
+	 * @param done number of records done
+	 * @param total the total number of records
+	 */
+	public void setDataPullStatus(long done, long total) {
+		User user = Context.getUserContext().getAuthenticatedUser();
+		if (user != null) {
+			user.setUserProperty("IITMLPullDone", String.valueOf(done));
+			user.setUserProperty("IITMLPullTotal", String.valueOf(total));
+			System.err.println("ITT ML - setting user properties: Done" + done + " and Total: " + total);
```
Comment (ojwanganto) [inline api/src/main/java/org/openmrs/module/kenyaemrml/util/MLDataExchange.java:416]: Should we remove this line? I guess you used it for debugging

File: omod/src/main/java/org/openmrs/module/kenyaemrml/fragment/controller/IitRiskScoreHistoryFragmentController.java [hunk 2]
```diff
@@ -81,6 +81,35 @@ public void stopDataPull(@SpringBean KenyaUiUtils kenyaUi, UiUtils ui) {
...
+				total = Long.parseLong(strTotal);
+			} catch(Exception ex) {}
+			double percent = 0.00;
+			if(total > 0.00 && done > 0.00) {
+				percent = ((done * 1.00/total * 1.00) * 100.0);
+			}
+			summary.put("done", done);
+			summary.put("total", total);
+			summary.put("percent", (int)Math.floor(percent));
+			//System.out.println("ITT ML - Got the pull status as: " + summary);
```
Comment (ojwanganto) [inline omod/src/main/java/org/openmrs/module/kenyaemrml/fragment/controller/IitRiskScoreHistoryFragmentController.java:109]: we can remove this line

File: omod/src/main/webapp/pages/dataManagement.gsp [hunk 3]
```diff
@@ -209,6 +231,47 @@ tr:nth-child(even) {background-color: #f2f2f2;}
             }).promise();
         }
 
+        // Check the status of the data pull
+        function fetchStatus() {
+            console.log('Checking fetch status!');
```
Comment (ojwanganto) [inline omod/src/main/webapp/pages/dataManagement.gsp:236]: remove this

File: omod/src/main/webapp/pages/dataManagement.gsp [hunk 4]
```diff
@@ -209,6 +231,47 @@ tr:nth-child(even) {background-color: #f2f2f2;}
...
+        function getFetchStatusAsync() {
+            let dfrd = jq.Deferred();
+            //The Task
+            ui.getFragmentActionAsJson('kenyaemrml', 'iitRiskScoreHistory', 'getStatusOfDataPull', {}, function (result) {
+                if(result) {
+                    //console.log('Success fetching pull status!');
+                    let statusDone = result.done;
+                    let statusTotal = result.total;
+                    let statusPercent = result.percent;
+                    //console.log('Got done as: ' + statusDone);
```
Comment (ojwanganto) [inline omod/src/main/webapp/pages/dataManagement.gsp:256]: remove the comments

**Review by ojwanganto — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/kenyaemrml/util/MLDataExchange.java (hunk 1 above)
Comment (PatrickWaweru) [reply api/src/main/java/org/openmrs/module/kenyaemrml/util/MLDataExchange.java]: Done

File: omod/src/main/java/org/openmrs/module/kenyaemrml/fragment/controller/IitRiskScoreHistoryFragmentController.java (hunk 2 above)
Comment (PatrickWaweru) [reply omod/src/main/java/org/openmrs/module/kenyaemrml/fragment/controller/IitRiskScoreHistoryFragmentController.java]: Done

File: omod/src/main/webapp/pages/dataManagement.gsp (hunk 3 above)
Comment (PatrickWaweru) [reply omod/src/main/webapp/pages/dataManagement.gsp]: Done

File: omod/src/main/webapp/pages/dataManagement.gsp (hunk 4 above)
Comment (PatrickWaweru) [reply omod/src/main/webapp/pages/dataManagement.gsp]: Done

**Pull request merged**